src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { IFetchComponent } from "@well-known-components/http-server"
+import { IConfigComponent, ILoggerComponent, IMetricsComponent } from "@well-known-components/interfaces"
+import { ISubgraphComponent, SubgraphResponse, Variables } from "./types"
+import { sleep } from "./utils"
+
+/**
+ * @public
+ */
+export async function createSubgraphComponent(
+  url: string,
+  components: createSubgraphComponent.NeededComponents
+): Promise<ISubgraphComponent> {
+  const { logs, metrics, config, fetch } = components
+
+  const logger = logs.getLogger("thegraph-port")
+
+  const RETRIES = (await config.getNumber("SUBGRAPH_COMPONENT_RETRIES")) || 3
+  const TIMEOUT = (await config.getNumber("SUBGRAPH_COMPONENT_QUERY_TIMEOUT")) || 5000
+  const BACKOFF = (await config.getNumber("SUBGRAPH_COMPONENT_BACKOFF")) || 500
+
+  async function executeQuery<T>(query: string, variables: Variables = {}, remainingAttempts?: number): Promise<T> {
+    logger.log(remainingAttempts === RETRIES ? `Querying subgraph ${url}` : `Retrying query to subgraph ${url}`)
```

Comment:
I'm thinking about the log priorities, should we use `info` here?
```suggestion
    logger.log(remainingAttempts === RETRIES ? `Querying subgraph ${url}` : `Retrying query to subgraph ${url}`)
```
Also, in the line 42, we're logging an error, I think we should use `log.error` there.